src/Shared/HandyControl_Shared/Controls/Attach/DataGridAttach.cs

```diff
@@ -338,19 +339,10 @@ public static void ClickBlankUnselectPropertyChanged(DependencyObject d, Depende
 
         private static void DataGrid_PreviewMouseDown(object sender, System.Windows.Input.MouseButtonEventArgs e)
         {
-            if (sender is DataGrid dataGrid)
+            if (sender is DataGrid dataGrid && e.OriginalSource is System.Windows.Controls.ScrollViewer)
```

Comment:
Do you test the code you write?
Now only the lower part of the program works and the empty spaces of the upper part do not work

Reply:
What do you think about using this button for this?
![2](https://user-images.githubusercontent.com/9213496/100469566-af370b00-30eb-11eb-9cbb-0d12101bb2f0.png)

Reply:
Only the lower part works is anticipated behavior, because this area is a `ScrollViewer`, the upper(header part) is not. 
The `SelectAll` button works well in my test.

![FWhEQ9U7xo](https://user-images.githubusercontent.com/4510984/100490625-a845e200-3157-11eb-82b5-70e187774190.gif)